### DISCUSSION TITLE
Generic way to show/hide arbitrary content on the page using react-aria

### DISCUSSION THREAD

**Discussion by smashercosmo** [adobe/react-spectrum#2588] [Q&A] [answered]
There are many common components that allows for showing/hiding some content, like tabs, accordions, modal dialogs and so on. But sometimes you just need something simpler, closer to html `<details>` tag. So, I'm wondering what would be the best way to implement something like that using the existing react-area hooks?

**Comment by reidbarber** [comment]
I don't think React Aria has exactly what you need at the moment, but I would recommend taking a look at implementing something similar to `useAccordionItem`, and removing collection-specific implementation details:

https://github.com/adobe/react-spectrum/blob/0571f1cee4565674086ea834327d0d59c093ea1f/packages/%40react-aria/accordion/src/useAccordion.ts#L36-L65

i.e. change state from TreeState to something with just `isExpanded: boolean, isDisabled: boolean`. Everything currently returned should probably work for your use case.

Maybe someone else can chime in if they're aware of an easier solution.

**Reply by smashercosmo** [reply]
Yeah, this is exactly what I was thinking about)

**Reply by reidbarber** [reply]
Sounds good! And for a state hook, you could copy `useOverlayTriggerState` as a starting point:

https://github.com/adobe/react-spectrum/blob/0571f1cee4565674086ea834327d0d59c093ea1f/packages/%40react-stately/overlays/src/useOverlayTriggerState.ts#L16-L46

**Reply by reidbarber** [reply, marked as answer]
Although, it might be better just to use our existing Accordion and contribute anything you find that is missing. Read up on benefits of the accordion pattern vs others: https://www.hassellinclusion.com/blog/accessible-accordions-part-2-using-details-summary/

**Reply by smashercosmo** [reply]
But, I'm wondering would react-spectrum team consider implementing some genetic hide/show content hook?

**Reply by snowystinger** [reply]
Can you explain your use case a bit more? or apprehension for using an Accordion with a single item? I think the Accordion approach is probably what would be used, as @reidbarber pointed it has some accessibility advantages.
